### ISSUE TITLE
CJS-to-AMD transpilation does not lift dependencies to factory function

### ISSUE THREAD

**Issue by rjgotten** [stealjs/transpile#132]
Given

```js
var foo = require('./foo');
var bar = require('./bar');
module.exports = [ foo, bar ];
```

The output when transpiled to AMD is

```js
define([
  'require',
  'exports',
  'module',
  './foo',
  './bar'
], function( require, exports, module ) {
  var foo = require('./foo');
  var bar = require('./bar');
  module.exports = [ foo, bar ];
});
```

Expected output, following AMD conventions instead of CJS-inside-AMD compatibility syntax, would be

```js
define([
  './foo',
  './bar'
], function( foo, bar ) {
  return [ foo, bar ];
});
```

**Comment by matthewp** [comment]
Does this present a problem some where? I thought all AMD loaders supported the CJS compat syntax, is that wrong?

**Comment by rjgotten** [comment]
> Does this present a problem some where?

Non-functionally speaking; AMD syntax proper offers better minification.

There _may_ also be issues with optimization / production-build tooling that attempts to analyze module source code to find CJS-style require calls and translate them, which in this case would give unexpected results. So there's also a potential compat problem there. (Not sure if that is still an issue, actually. But it definitely used to be.)
